Core/Core/Data/Model/Data_PrimaryEnrollment.swift

```diff
@@ -52,6 +52,25 @@ public extension DataLayer {
             case courseAssignments = "course_assignments"
         }
         
+        public init(from decoder: Decoder) throws {
+            let values = try decoder.container(keyedBy: CodingKeys.self)
+            
+            if let dateString = try? values.decode(String.self, forKey: .auditAccessExpires) {
+                auditAccessExpires = Date(iso8601: dateString)
+            } else {
+                auditAccessExpires = nil
+            }
+            created = try? values.decode(String.self, forKey: .created)
+            mode = try? values.decode(String.self, forKey: .mode)
+            isActive = try? values.decode(Bool.self, forKey: .isActive)
+            course = try? values.decode(DashboardCourse.self, forKey: .course)
+            certificate = try? values.decode(Certificate.self, forKey: .certificate)
+            courseModes = try? values.decode([CourseMode].self, forKey: .courseModes)
+            courseStatus = try? values.decode(CourseStatus.self, forKey: .courseStatus)
+            progress = try? values.decode(CourseProgress.self, forKey: .progress)
+            courseAssignments = try? values.decode(CourseAssignments.self, forKey: .courseAssignments)
+        }
```

Comment:
If the `decoder` is only introduced for `audit_access_expires`, that can be omitted by changing the `auditAccessExpires` type from Date? to String?